src/wayland/shell/xdg/mod.rs

```diff
@@ -567,13 +881,81 @@ where
     /// Send a configure event to this toplevel surface to suggest it a new configuration
     ///
     /// The serial of this configure will be tracked waiting for the client to ACK it.
-    pub fn send_configure(&self, cfg: ToplevelConfigure) {
-        if !self.alive() {
-            return;
+    pub fn send_configure(&self) {
+        // TODO: Handle the idle scheduling of the configure here?
```

Comment:
No. The method `send_configure` should do the actual sending. It's up to the user to schedule invoking this method from an idle callback if they desire so.